File: app/src/main/java/com/test/kabak/openweather/core/storage/CurrentWeatherDao.java
package com.test.kabak.openweather.core.storage;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface CurrentWeatherDao {
    @Insert(onConflict = REPLACE)
    void insertAll(List<CurrentWeather> weatherArray);

    @Query("SELECT * FROM CurrentWeather where cityId = :cityId")
    CurrentWeather getById(String cityId);
}

File: app/src/main/java/com/test/kabak/openweather/core/storage/CurrentWeatherDao.kt
package com.test.kabak.openweather.core.storage

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy.REPLACE
import androidx.room.Query

@Dao
interface CurrentWeatherDao {
    @Insert(onConflict = REPLACE)
    suspend fun insertAll(weatherArray: List<CurrentWeather>)

    @Insert(onConflict = REPLACE)
    suspend fun insert(weatherArray: CurrentWeather)

    @Query("SELECT * FROM CurrentWeather where cityId = :cityId")
    suspend fun getById(cityId: String): CurrentWeather?
}